s3/replication/common/tests/system/test_put_bucket_replication.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2021 Seagate Technology LLC and/or its Affiliates
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+#
+# s3 get-bucket-replication api test script
+# Usage : python3 test_s3_get_bucket_replication.py
+#
+
+import os
+import subprocess
+import fileinput
+from config import Config
+
+
+def main():
+
+    config = Config()
+
+    # CONFIG OPTIONS
+    bucket_name = config.source_bucket_name
+    dest_bucket_name = config.target_bucket_name
+    object_name_prefix = config.object_name_prefix
+    iam_role = config.iam_role
+
+    # Create temp replication policy file.
+    subprocess.call(['cp',
+                     './tests/system/config/replication_policy_sample.json',
+                     'temp_policy.json'])
+
+    matches = ['ROLE_ID', 'REPLICATION_ENABLED_BUCKET']
+
+    # Read policy and make replacements based on config options.
+    with fileinput.FileInput('temp_policy.json', inplace=True) as file:
+
+        # Read each line and match the pattern and do replacement.
+        for line in file:
+            if all(x in line for x in matches):
```

Comment:
this is not readable, please find alternative.

Reply:
One option was to do if-else statement which will be like adding two conditions. this particular statement is very common for the python code as per my search. I guess we can keep it.

or as in the example rule template, both the patterns which we are trying to match stays on the same line I can do like -
if 'ACCOUNT_ID' in line:
         line = re.sub('(ACCOUNT_ID)', str(iam_role), line)
         line = re.sub('(REPLICATION_ENABLED_BUCKET)', bucket_name, line)
but seems hardcoded and  specific to this rule template only.